backend/app/services/adapters/task_kinds.py

```
@@ -1810,6 +1810,22 @@ def delete_task(self, db: Session, *, task_id: int, user_id: int) -> None:
 
+        # Clean up long-term memories associated with this task (fire-and-forget)
+        # This runs in background and doesn't block task deletion
+        from app.services.memory import get_memory_manager
+
+        memory_manager = get_memory_manager()
+        if memory_manager.is_enabled:
+            asyncio.create_task(
+                memory_manager.cleanup_task_memories(
+                    user_id=task.user_id, task_id=task_id
+                )
+            )
```

Ruff (0.14.11)

Store a reference to the return value of `asyncio.create_task`

(RUF006)